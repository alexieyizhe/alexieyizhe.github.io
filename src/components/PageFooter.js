import React from "react";
import styled from "styled-components";
import { animateScroll } from "react-scroll";
import SVGDrawIcon from "./SVGDrawIcon.js";
import Icon from "./Icon.js";
import { mediaSize } from "../data/configOptions.js";
import Link from "gatsby-link";

const FooterContainer = styled.span`
  margin: 10% 5vw;
  width: 30%;
  margin-left: auto;
  margin-right: auto;
  display: grid;
  grid-template-columns: 1fr 1fr 1.5fr 1fr 1fr;
  grid-template-rows: 2fr auto;
  justify-items: center;
  align-items: center;
  grid-template-areas: "contactIcon resumeIcon topIcon githubIcon linkedinIcon"
                       "copyright copyright copyright copyright copyright";


 ${mediaSize.tablet`
   width: 60%;
 `}

 ${mediaSize.phone`
   width: 90%;
 `}
`;

const PageFooter = () => (
  <FooterContainer>
    <a href="mailto:alex@alexieyizhe.me" target="_blank" style={{gridArea: 'contactIcon'}}>
      <SVGDrawIcon>
        <Icon name="paper_plane" size="2vh" color="#80D07F" />
      </SVGDrawIcon>
    </a>
    <Link to="/resume" style={{gridArea: 'resumeIcon'}}>
      <SVGDrawIcon>
        <Icon name="file" size="2vh" color="#DE7947" />
      </SVGDrawIcon>
    </Link>
    <span style={{gridArea: 'topIcon'}} onClick={() => animateScroll.scrollToTop()}>
      <SVGDrawIcon>
        <Icon name="up_arrow" size="4vh" color="#000" />
      </SVGDrawIcon>
    </span>
    <a href="https://github.com/alexieyizhe" target="_blank" style={{gridArea: 'githubIcon'}}>
      <SVGDrawIcon>
        <Icon name="github" size="2vh" color="#B29CE7" />
      </SVGDrawIcon>
    </a>
    <a href="https://www.linkedin.com/in/alexieyizhe/" target="_blank" style={{gridArea: 'linkedinIcon'}}>
      <SVGDrawIcon>
        <Icon name="linkedin"  size="2vh" color="#4092DE"/>
      </SVGDrawIcon>
    </a>
    <p style={{gridArea: 'copyright'}}>
      Â© alex xie, mmxviii.
    </p>
  </FooterContainer>
);

export default PageFooter;
