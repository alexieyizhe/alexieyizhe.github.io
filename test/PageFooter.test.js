import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import PageFooter from '../src/components/PageFooter';


describe('Page Footer', () => {
  let footer;

  beforeEach(() => {
    footer = shallow(
      <MemoryRouter>
        <PageFooter className="pageFooter" />
      </MemoryRouter>
    ).children();
  });

  /* react-mt-svg-lines breaks renderer
  it('should render the footer', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <PageFooter className="pageFooter" />
        </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  */

  it('should render the footer', () => {
    expect(footer).toMatchSnapshot();
  });

  describe('default state', () => {
    it('should be defined', () => {
      expect(footer).toBeDefined();
    });
  });

  it.skip('should trigger scroll to top', () => {
    footer = mount(
      <MemoryRouter>
        <PageFooter className="pageFooter" />
      </MemoryRouter>
    ).children();
    footer.find('span#scrollToTopTrigger').simulate('click');
  })

});
