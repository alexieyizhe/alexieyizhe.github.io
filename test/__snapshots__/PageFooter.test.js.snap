// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Footer should render the footer 1`] = `
<PageFooter
  className="pageFooter"
/>
`;
